app/models/base_facility.rb

```diff
@@ -201,6 +204,20 @@ def build_distance_result_set(lat, long, type, services, ids, limit = nil)
     end
     # rubocop:enable Metrics/ParameterLists
 
+    def zip_query(params)
+      # TODO: allow user to set distance from zip
+      zip_plus0 = params[:zip][0...5]
+      zcta = CSV.read(Rails.root.join('modules', 'va_facilities', '2018_Gaz_zcta_national.txt'), col_sep: "\t")
```

Comment:
Would it be worth reading this one at startup and putting it in a constant? @erluti recommended using an initializer to me. You would do this reading work in `/modules/va_facilities/config/initializers/zcta.rb` and then reference the constant you create here.